src/detect-ipproto.c

```diff
@@ -121,20 +122,23 @@ static DetectIPProtoData *DetectIPProtoParse(const char *optstr)
 
     /* Protocol name/number */
     if (!isdigit((unsigned char)*(args[1]))) {
-        struct protoent *pent = getprotobyname(args[1]);
-        if (pent == NULL) {
-            SCLogError(SC_ERR_INVALID_VALUE, "Malformed protocol name: %s",
-                       str_ptr);
+        int proto = SCGetProtoByName(args[1]);
+        if (-1 == proto) {
+            SCLogError(SC_ERR_INVALID_VALUE, "Malformed protocol name: \"%s\"", str_ptr);
             goto error;
         }
-        data->proto = (uint8_t)pent->p_proto;
+        data->proto = (uint8_t)proto;
     }
     else {
         if (StringParseUint8(&data->proto, 10, 0, args[1]) <= 0) {
             SCLogError(SC_ERR_INVALID_VALUE, "Malformed protocol number: %s",
                        str_ptr);
             goto error;
         }
+        if (!SCProtoNameValid((uint16_t)data->proto)) {
```

Comment:
will this not limit our support for pretty much no reason?
`ip_proto:144` would be rejected as bad input even though it could just work?

Reply:
Good point. Will update.